### PR TITLE
Documentation Updates: docker-compose examples

### PR DESCRIPTION
**Pull request by karnthis**
### What does this PR do?

I have updated all the occurrences of `image: traefik` to use the Alpha tag `image: traefik:v2.0`. I also corrected the command from `--docker` to `--providers.docker`. Lastly, I renamed `Getting Started` > `Configuration Overview` to `Configuration Introduction`. This is because there is also `Configuration Discovery` > `Overview`, and as both have a large identical image occupying the bulk of the opening page they are easy to confuse I feel. This last change is more of a proposal/suggestion.

### Motivation

This PR came to life because I was discovering challenges with converting my Traefik v1 docker-compose files to v2. This turned out to be because the v2 Alpha docs appeared to still be using sample docker-compose code from v1.

### CHANGES

- Replaced all documentation occurrences of `image: traefik` (4) with `image: traefik:v2.0`
- Replaced all documentation occurrences of `--docker` (1) with `--providers.docker`
- Renamed page under `Getting Started` from `Configuration Overview` to `Configuration Introduction` for clarity

### Additional Notes

I am very excited about the new changes, and I am looking forward to have time to dig in deeper, learn the changes, and maybe update some more documentation.

### CODE REVIEW

**Comment by karnthis** [comment]
I think I tried to merge this into the wrong branch. I thought I was merging back into what I branched from but it looks like I did not.

**Comment by karnthis** [comment]
It looks like I forked from the release tag v2.0.0-alpha2, and because it is a tag not a branch I can't merge back to it. How should I proceed?

File: docs/content/getting-started/quick-start.md [hunk 1]
```diff
@@ -17,8 +17,8 @@ version: '3'
 
 services:
   reverse-proxy:
-    image: traefik # The official Traefik docker image
-    command: --api --docker # Enables the web UI and tells Traefik to listen to docker
+    image: traefik:v2.0 # The official v2.0 Alpha Traefik docker image
```
Comment (ldez) [inline docs/content/getting-started/quick-start.md]:
```suggestion
    image: traefik:v2.0 # The official v2.0 Traefik docker image
```

File: docs/content/providers/docker.md [hunk 2]
```diff
@@ -124,7 +124,7 @@ Traefik requires access to the docker socket to get its dynamic configuration.
     services:
 
       traefik:
-         image: traefik
+         image: traefik:v2.0 # The official v2.0 Alpha Traefik docker image
```
Comment (ldez) [inline docs/content/providers/docker.md]:
```suggestion
         image: traefik:v2.0 # The official v2.0 Traefik docker image
```

File: docs/content/routing/entrypoints.md [hunk 3]
```diff
@@ -64,7 +64,7 @@ Entrypoints are part of the [static configuration](../getting-started/configurat
         
         ```yaml
         traefik:
-            image: traefik
+            image: traefik:v2.0 # The official v2.0 Alpha Traefik docker image
```
Comment (ldez) [inline docs/content/routing/entrypoints.md]:
```suggestion
            image: traefik:v2.0 # The official v2.0 Traefik docker image
```

File: docs/content/routing/entrypoints.md [hunk 4]
```diff
@@ -74,7 +74,7 @@ Entrypoints are part of the [static configuration](../getting-started/configurat
...
```
Comment (ldez) [inline docs/content/routing/entrypoints.md]:
```suggestion
            image: traefik:v2.0 # The official v2.0 Traefik docker image
```

**Comment by karnthis** [comment]
> Thanks, could you rebase?

Could you clarify your requests? Removing Alpha is fine, but I do not understand why `command: --api --docker` should be removed. 

My understanding is:
- `--docker` is required for traefik to have access to the docker daemon.
- `--api` is what enables the Traefik API, which is what the quick start uses to showcase operation.

**Comment by ldez** [comment]
could you update your PR (rebase or merge like want) to get changes from the v2.0 branch.

**Comment by ldez** [comment]
> Could you clarify your requests? Removing Alpha is fine, but I do not understand why command: --api --docker should be removed.

It's just the diff in Github :smile:

**Comment by karnthis** [comment]
Ok great thanks for the clarification. I will rebase tonight and post again here.

**Comment by karnthis** [comment]
Changes requested has been made. It looks like there is a conflict with the Quick Start sample code.

**Pull request merged**